luminoth/train.py

```diff
@@ -88,6 +90,43 @@ def run(config, target='', cluster_spec=None, is_chief=True, job_name=None,
                 grads_and_vars, global_step=global_step
             )
 
+        # Create custom init for slots in optimizer, as we don't save them to
+        # our checkpoints. An example of slots in an optimizer are the Momentum
+        # variables in MomentumOptimizer.
+        slot_variables = [
+            optimizer.get_slot(var, name)
+            for name in optimizer.get_slot_names()
+            for var in trainable_vars
+        ]
+        slot_init = tf.variables_initializer(
+            slot_variables,
+            name='optimizer_slots_initializer'
+        )
+
+        # Create saver for saving/restoring model
```

Comment:
Clarify why this is done. Remove slot variables to avoid storing momentum in checkpoint files for higher space efficiency, or something like that.

Reply:
I explained that we were not saving the slot variables about 10 lines before this one. I think that explanation plus the `set(tf.global_variables()) - set(slot_variables)` in the saver should make it quite clear?